ai_diffusion/model.py

```diff
@@ -622,6 +633,89 @@ def _finish_job(self, job: Job, event: ClientEvent):
             self.jobs.notify_cancelled(job)
             self.progress = 0
 
+    def _apply_resize_command(self, cmd: ResizeCommand, job: Job):
+        """Record a requested canvas resize for this job.
+
+        The actual resize is applied when showing a preview or applying the result.
+        """
+        try:
+            bounds = job.params.bounds
+            job.params.bounds = Bounds(bounds.x, bounds.y, cmd.width, cmd.height)
+            job.params.metadata["resize_canvas"] = {"width": cmd.width, "height": cmd.height}
+        except Exception as e:
+            log.warning(f"Failed to store resize command from custom workflow: {e}")
+
+    def _get_resize_for_params(self, params: JobParams) -> tuple[int, int] | None:
+        resize = params.metadata.get("resize_canvas")
+        if not isinstance(resize, dict):
+            return None
+
+        try:
+            width = int(resize.get("width", 0))
+            height = int(resize.get("height", 0))
+        except Exception:
+            return None
+
+        if width <= 0 or height <= 0:
+            return None
+
+        return width, height
+
+    def _preview_canvas_size(self, params: JobParams):
```

Comment:
I'm not sure it's a good idea to resize on preview. I know you will only get a partial preview otherwise, but for high-res documents with a lot of layers, resize can take several seconds. Waiting seconds when toggling preview sounds annoying. You can always apply and undo, which is maybe better in this case.